From 897875a893641ca627d553506058297d2c0493c8 Mon Sep 17 00:00:00 2001
From: Erin Park <erin.park@intel.com>
Date: Wed, 23 Oct 2019 10:54:48 +0000
Subject: [PATCH 2/2] R3 install DLDT headers, libs, sample apps

 -Libraries to /usr/lib64
 -Headers to /usr/include
 -Python library to /usr/lib/python3.7/site-packages
 -plugins.xml to /usr/share/openvino/inference-engine
 -Create targets_cpu_extension.cmake

Signed-off-by: Hui Chun Ong <hui.chun.ong@intel.com>
Signed-off-by: Choong Yin Thong <yin.thong.choong@intel.com>
---
 inference-engine/CMakeLists.txt               |  2 ++
 inference-engine/cmake/config.cmake.in        |  1 +
 .../ie_bridges/python/CMakeLists.txt          |  2 +-
 .../openvino/inference_engine/CMakeLists.txt  |  4 ++-
 inference-engine/samples/CMakeLists.txt       |  9 +------
 .../common/format_reader/CMakeLists.txt       |  2 ++
 inference-engine/src/CMakeLists.txt           |  4 ++-
 inference-engine/src/extension/CMakeLists.txt |  5 +++-
 .../src/inference_engine/CMakeLists.txt       | 25 ++++++++++++++++---
 .../src/inference_engine/ie_util_internal.cpp |  7 +++---
 10 files changed, 42 insertions(+), 19 deletions(-)

diff --git a/inference-engine/CMakeLists.txt b/inference-engine/CMakeLists.txt
index a282cb439186898b057f84eead5fcee3f2c46ba6..abb4de4126e23cff42f4fcfa45bb5689a684c3b7 100644
--- a/inference-engine/CMakeLists.txt
+++ b/inference-engine/CMakeLists.txt
@@ -70,3 +70,5 @@ if (ENABLE_PYTHON)
 endif()
 
 add_cpplint_report_target()
+
+install(DIRECTORY samples DESTINATION share/doc/inference_engine)
diff --git a/inference-engine/cmake/config.cmake.in b/inference-engine/cmake/config.cmake.in
index ebe82ee90a76b1e7394536bda2b30f91eaca7808..0d7e6a4730f4920341a77f9e07bf7d6883beee4a 100644
--- a/inference-engine/cmake/config.cmake.in
+++ b/inference-engine/cmake/config.cmake.in
@@ -7,6 +7,7 @@ if(DEFINED IE_MAIN_SOURCE_DIR AND TARGET inference_engine)
     set(InferenceEngine_LIBRARIES inference_engine)
 else()
     include("${CMAKE_CURRENT_LIST_DIR}/targets.cmake")
+    include("${CMAKE_CURRENT_LIST_DIR}/targets_cpu_extension.cmake")
     if(NOT WIN32)
         set_target_properties(IE::inference_engine PROPERTIES INTERFACE_COMPILE_OPTIONS "-Wno-error=deprecated-declarations")
     endif()
diff --git a/inference-engine/ie_bridges/python/CMakeLists.txt b/inference-engine/ie_bridges/python/CMakeLists.txt
index bba853fb289ecc0d57f82350282f2d02038f74cb..444cce9d9ea7c98987bc5aae2d423c681e5f3413 100644
--- a/inference-engine/ie_bridges/python/CMakeLists.txt
+++ b/inference-engine/ie_bridges/python/CMakeLists.txt
@@ -53,4 +53,4 @@ endif()
 
 set (PYTHON_BRIDGE_SRC_ROOT ${CMAKE_CURRENT_SOURCE_DIR})
 add_subdirectory (src/openvino/inference_engine)
-add_subdirectory (src/openvino/tools/statistics_collector)
+#add_subdirectory (src/openvino/tools/statistics_collector)
diff --git a/inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt b/inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt
index 1f46013b4be38f300ebac9f51ebe79579247416c..d6cd112840b183b44ba6448f005aa286a65f5937 100644
--- a/inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt
+++ b/inference-engine/ie_bridges/python/src/openvino/inference_engine/CMakeLists.txt
@@ -29,4 +29,6 @@ ADD_CUSTOM_COMMAND (TARGET ${TARGET_NAME}
     COMMAND ${CMAKE_COMMAND} -E copy ${PYTHON_BRIDGE_SRC_ROOT}/src/openvino/inference_engine/__init__.py ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/__init__.py
     COMMAND ${CMAKE_COMMAND} -E copy ${PYTHON_BRIDGE_SRC_ROOT}/requirements.txt ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../../requirements.txt
     COMMAND ${CMAKE_COMMAND} -E copy ${PYTHON_BRIDGE_SRC_ROOT}/src/openvino/__init__.py ${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/../__init__.py
-)
\ No newline at end of file
+)
+
+install(DIRECTORY ${PYTHON_BRIDGE_OUTPUT_DIRECTORY} DESTINATION lib/${PYTHON_VERSION}/site-packages/)
diff --git a/inference-engine/samples/CMakeLists.txt b/inference-engine/samples/CMakeLists.txt
index 884b0a6374d364656b4ebaece66d77349d7cddd9..25f9f23492a1c7543d81cbdfa2e4413807eea6ff 100644
--- a/inference-engine/samples/CMakeLists.txt
+++ b/inference-engine/samples/CMakeLists.txt
@@ -39,14 +39,6 @@ if(NOT(UNIX))
     set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER})
     set (LIBRARY_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER})
     set (LIBRARY_OUTPUT_PATH ${LIBRARY_OUTPUT_DIRECTORY}) # compatibility issue: linux uses LIBRARY_OUTPUT_PATH, windows uses LIBRARY_OUTPUT_DIRECTORY
-else ()
-    set (CMAKE_LIBRARY_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE}/lib)
-    set (CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE}/lib)
-    set (CMAKE_COMPILE_PDB_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE})
-    set (CMAKE_PDB_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE})
-    set (CMAKE_RUNTIME_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE})
-    set (LIBRARY_OUTPUT_DIRECTORY ${IE_MAIN_SAMPLES_DIR}/${BIN_FOLDER}/${CMAKE_BUILD_TYPE}/lib)
-    set (LIBRARY_OUTPUT_PATH ${LIBRARY_OUTPUT_DIRECTORY}/lib)
 endif()
 
 if (WIN32)
@@ -188,6 +180,7 @@ macro(ie_add_sample)
     if(COMMAND add_cpplint_target AND NOT IE_SAMPLE_EXCLUDE_CPPLINT)
         add_cpplint_target(${IE_SAMPLE_NAME}_cpplint FOR_TARGETS ${IE_SAMPLE_NAME})
     endif()
+    install(TARGETS ${IE_SAMPLE_NAME} DESTINATION bin)
 endmacro()
 
 
diff --git a/inference-engine/samples/common/format_reader/CMakeLists.txt b/inference-engine/samples/common/format_reader/CMakeLists.txt
index a8c9caf0b5f6764cfffbabfad2b6c1e41965a1f2..a4109f31ea1af65a7af6280af0715e36f84ddde0 100644
--- a/inference-engine/samples/common/format_reader/CMakeLists.txt
+++ b/inference-engine/samples/common/format_reader/CMakeLists.txt
@@ -40,3 +40,5 @@ set_target_properties(${TARGET_NAME} PROPERTIES COMPILE_PDB_NAME ${TARGET_NAME})
 
 export(TARGETS ${TARGET_NAME} NAMESPACE IE::
        APPEND FILE "${CMAKE_BINARY_DIR}/targets_developer.cmake")
+
+install(TARGETS ${TARGET_NAME} LIBRARY DESTINATION lib64)
diff --git a/inference-engine/src/CMakeLists.txt b/inference-engine/src/CMakeLists.txt
index 63fda2ada6f380afa2304c2ce79aeb373bbabf11..3a4bc48c37d99424bd195cddf605a43c985b172e 100644
--- a/inference-engine/src/CMakeLists.txt
+++ b/inference-engine/src/CMakeLists.txt
@@ -37,8 +37,10 @@ set(InferenceEngine_SRC_DIRS ${CMAKE_SOURCE_DIR}/src)
 function(set_target_cpu_flags TARGET_NAME)
 endfunction()
 
-add_subdirectory(extension EXCLUDE_FROM_ALL)
+add_subdirectory(extension)
 add_library(IE::ie_cpu_extension ALIAS ie_cpu_extension)
 
 file(GLOB_RECURSE EXTENSION_SOURCES extension/*.cpp extension/*.hpp extension/*.h)
 add_cpplint_target(ie_cpu_extension_cpplint FOR_SOURCES ${EXTENSION_SOURCES})
+
+install(DIRECTORY extension DESTINATION share/doc/inference_engine)
diff --git a/inference-engine/src/extension/CMakeLists.txt b/inference-engine/src/extension/CMakeLists.txt
index 799a62de4ee642986aac7e4b7ab2265f27110ff1..f7d3bebf194a7baa0691d12b22c8ba8ada46a00e 100644
--- a/inference-engine/src/extension/CMakeLists.txt
+++ b/inference-engine/src/extension/CMakeLists.txt
@@ -49,7 +49,7 @@ set_target_properties(${TARGET_NAME} PROPERTIES OUTPUT_NAME "cpu_extension")
 
 target_link_libraries(${TARGET_NAME} PRIVATE ${InferenceEngine_LIBRARIES})
 
-target_include_directories(${TARGET_NAME} PUBLIC ${CMAKE_CURRENT_SOURCE_DIR})
+target_include_directories(${TARGET_NAME} PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>)
 set_target_properties(${TARGET_NAME} PROPERTIES COMPILE_PDB_NAME ${TARGET_NAME})
 
 set_target_cpu_flags(${TARGET_NAME})
@@ -61,3 +61,6 @@ if (IE_MAIN_SOURCE_DIR)
     export(TARGETS ${TARGET_NAME} NAMESPACE IE:: APPEND FILE "${CMAKE_BINARY_DIR}/targets_developer.cmake")
 endif()
 
+set(ConfigPackageLocation lib64/cmake/InferenceEngine)
+install(TARGETS ${TARGET_NAME} EXPORT ${TARGET_NAME} LIBRARY DESTINATION lib64)
+install(EXPORT ${TARGET_NAME} DESTINATION ${ConfigPackageLocation} NAMESPACE IE:: FILE targets_cpu_extension.cmake)
diff --git a/inference-engine/src/inference_engine/CMakeLists.txt b/inference-engine/src/inference_engine/CMakeLists.txt
index 2604c32b62f31347e0b56dd03f3d67255c66592c..1332ed2c1a94c3df53d36fffc4a20a876cb964c2 100644
--- a/inference-engine/src/inference_engine/CMakeLists.txt
+++ b/inference-engine/src/inference_engine/CMakeLists.txt
@@ -110,8 +110,11 @@ if(WIN32)
 endif()
 
 # Properties->C/C++->General->Additional Include Directories
-target_include_directories(${TARGET_NAME} PUBLIC ${PUBLIC_HEADERS_DIR}
-                                          PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}")
+target_include_directories(${TARGET_NAME} PUBLIC
+                                          $<BUILD_INTERFACE:${IE_MAIN_SOURCE_DIR}/include>
+                                          $<INSTALL_INTERFACE:include/${TARGET_NAME}>
+                                          PRIVATE "${CMAKE_CURRENT_SOURCE_DIR}"
+                                          PRIVATE "${IE_MAIN_SOURCE_DIR}/src/dumper")
 
 target_include_directories(${TARGET_NAME} SYSTEM PRIVATE "${IE_MAIN_SOURCE_DIR}/thirdparty/pugixml/src")
 target_include_directories(${TARGET_NAME} SYSTEM PRIVATE "${IE_MAIN_SOURCE_DIR}/thirdparty/ngraph/src")
@@ -137,7 +140,9 @@ add_library(${TARGET_NAME}_s STATIC
 set_ie_threading_interface_for(${TARGET_NAME}_s)
 
 # Properties->C/C++->General->Additional Include Directories
-target_include_directories(${TARGET_NAME}_s PUBLIC  "${PUBLIC_HEADERS_DIR}"
+target_include_directories(${TARGET_NAME}_s PUBLIC
+                                            $<BUILD_INTERFACE:${IE_MAIN_SOURCE_DIR}/include>
+                                            $<INSTALL_INTERFACE:include/${TARGET_NAME}>
                                             PUBLIC  "${CMAKE_CURRENT_SOURCE_DIR}"
                                             PRIVATE "${IE_MAIN_SOURCE_DIR}/src/dumper")
 
@@ -173,7 +178,8 @@ target_link_libraries(${TARGET_NAME}_s PRIVATE fluid
 add_cpplint_target(${TARGET_NAME}_cpplint FOR_TARGETS ${TARGET_NAME})
 
 ie_register_plugins(MAIN_TARGET ${TARGET_NAME}
-                    POSSIBLE_PLUGINS MultiDevicePlugin HeteroPlugin clDNNPlugin dliaPlugin GNAPlugin MKLDNNPlugin myriadPlugin HDDLPlugin)
+                       $<BUILD_INTERFACE:${IE_MAIN_SOURCE_DIR}/include>
+                       $<INSTALL_INTERFACE:include/${TARGET_NAME}>POSSIBLE_PLUGINS clDNNPlugin dliaPlugin GNAPlugin MKLDNNPlugin myriadPlugin)
 
 # export targets
 export(TARGETS ${TARGET_NAME} NAMESPACE IE:: FILE "${CMAKE_BINARY_DIR}/targets.cmake")
@@ -204,3 +210,14 @@ configure_file(
     "${CMAKE_SOURCE_DIR}/cmake/share/InferenceEngineConfig-version.cmake.in"
     "${CMAKE_BINARY_DIR}/InferenceEngineDeveloperPackageConfig-version.cmake"
     COPYONLY)
+
+set(ConfigPackageLocation lib64/cmake/InferenceEngine)
+install(TARGETS ${TARGET_NAME} EXPORT ${TARGET_NAME} LIBRARY DESTINATION lib64)
+install(EXPORT ${TARGET_NAME} DESTINATION ${ConfigPackageLocation} NAMESPACE IE:: FILE targets.cmake)
+install(DIRECTORY ${PUBLIC_HEADERS_DIR}/ DESTINATION include/${TARGET_NAME} FILES_MATCHING PATTERN "*.h" PATTERN "*.hpp")
+install(FILES
+    "${CMAKE_BINARY_DIR}/InferenceEngineConfig.cmake"
+    "${CMAKE_BINARY_DIR}/InferenceEngineConfig-version.cmake"
+    DESTINATION ${ConfigPackageLocation})
+
+install(FILES "$<TARGET_FILE_DIR:${TARGET_NAME}>/plugins.xml" DESTINATION /usr/share/openvino/inference-engine)
\ No newline at end of file
diff --git a/inference-engine/src/inference_engine/ie_util_internal.cpp b/inference-engine/src/inference_engine/ie_util_internal.cpp
index 1a2f0784607028b5056937219602c621ed2b792a..b2668a9375453b3216fcffd1d8c11767ec6f9f91 100644
--- a/inference-engine/src/inference_engine/ie_util_internal.cpp
+++ b/inference-engine/src/inference_engine/ie_util_internal.cpp
@@ -760,9 +760,10 @@ std::string getIELibraryPath() {
     GetModuleFileNameA(hm, (LPSTR)ie_library_path, sizeof(ie_library_path));
     return getPathName(ie_library_path);
 #else
-    Dl_info info;
-    dladdr(reinterpret_cast<void *>(getIELibraryPath), &info);
-    return getPathName(info.dli_fname);
+    //Dl_info info;
+    //dladdr(reinterpret_cast<void *>(getIELibraryPath), &info);
+    //return getPathName(info.dli_fname);
+    return  "/usr/share/openvino/inference-engine";
 #endif
 }
 
-- 
2.23.0

